src/tests/unit/tests/injected/client-store-listener.test.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { IMock, It, Mock, MockBehavior, Times } from 'typemoq';
+
+import { BaseClientStoresHub } from '../../../../common/stores/base-client-stores-hub';
+import { ClientStoreListener, TargetPageStoreData } from '../../../../injected/client-store-listener';
+
+describe('ClientStoreListener', () => {
+    describe('initialize', () => {
+        let testSubject: ClientStoreListener;
+        let storeHubMock: IMock<BaseClientStoresHub<TargetPageStoreData>>;
+        let onReadyToExecuteVisualizationUpdatesMock: IMock<(storeData: TargetPageStoreData) => void>;
+        let triggerOnChange: () => void;
+
+        beforeEach(() => {
+            storeHubMock = Mock.ofType<BaseClientStoresHub<TargetPageStoreData>>(null, MockBehavior.Strict);
```

Comment:
I think typemoq docs mention `undefined` for the first parameter will make use of es6 `Proxy` class and this is what we've been doing in general. Is there any specific reason why in here is `null`? Not that it matter that much except for the consistency of it

Reply:
> This looks good but I left some minor comments.
> I'm also assuming there will be another PR to hook up this new classes as I don't see then being use neither the `ClientViewController` being updated!
> Good refactor.

Yep! Didn't want a massive PR 😄